File: app/src/main/java/com/yeputra/moviecatalogue/adapter/VPagerAdapter.java
package com.yeputra.moviecatalogue.adapter;


import com.yeputra.moviecatalogue.model.VPager;
import java.util.List;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * Created by yovi.putra
 *    on 03/Mar/2019 17:22
 * Company SIEMO - PT. Multipolar Technology, Tbk
 */
public class VPagerAdapter extends FragmentPagerAdapter {
    private List<VPager> fragments;
    private FragmentManager fmManager;

    public VPagerAdapter(List<VPager> fragments, FragmentManager fm) {
        super(fm);
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position).getFragment();
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return fragments.get(position).getTitle();
    }
}

File: app/src/main/java/com/yeputra/moviecatalogue/adapter/VPagerAdapter.kt
package com.yeputra.moviecatalogue.adapter


import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.yeputra.moviecatalogue.model.VPager

/**
 * Created by yovi.putra
 * on 03/Mar/2019 17:22
 * Company SIEMO - PT. Multipolar Technology, Tbk
 */
class VPagerAdapter(
        private val fragments: List<VPager>,
        fm: FragmentManager
) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragments[position].fragment
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragments[position].title
    }
}
